### ISSUE TITLE
Need to know number of http requests "evhttp_request" in one connection "evhttp_connection"

### ISSUE THREAD

**Issue by ochinchina** [libevent/libevent#657]
I have a case need to send only one request in one connection because of concurrent request to the server. So I need to know number of requests pending on a connection, if there is no any requests on the connection, I will put the request on this connection, if there any requests in the connection, need to find a new free connection for the request.

**Comment by azat** [comment]
> So I need to know number of requests pending on a connection

Hi,

There is no such API right now, and I'm not sure that this can be
useful.

Since you still need to track the moment when the request was removed
from the connection requests queue, and there is no easy/clean way to do
this right now. Sure you can add a timer and check every connection you
have but I don't think that this is a good idea.

I think that you should implement this in you user-specific code instead
of trying to force library do this for you.

But if you still not convinced than you do this by your own (for example
this can be implemented as something generic like rate-limiting API for
requests per-connection and this is the thing that can be merged into
upstream per se, i.e. like #592), and here is a pull that can you take a
look at to start -- #591.
Though I still think that this can be implemented on top of http API
using user-specific abstractions like request/connection pool.

Regards,
Azat.
